src/cif.cc

```diff
@@ -60,12 +82,61 @@ structure_from_doc(const gemmi::cif::Document &doc,
                 continue;
             }
 
-            auto model = atoi(site[11].c_str());
+            if (models.count(std::stoi(site[11])) == 0) continue;
+            
+            freesasa_cif_atom atom = {
+                .group_PDB = site[0].c_str(),
+                .auth_asym_id = site[1][0],
+                .auth_seq_id = site[2].c_str(),
+                .pdbx_PDB_ins_code = site[3].c_str(),
+                .auth_comp_id = site[4].c_str(),
+                .auth_atom_id = site[5].c_str(),
+                .label_alt_id = site[6].c_str(),
+                .type_symbol = site[7].c_str(),
+                .Cartn_x = atof(site[8].c_str()),
+                .Cartn_y = atof(site[9].c_str()),
+                .Cartn_z = atof(site[10].c_str())};
+
+            auto currentAltId = site[6][0];
+
+            if (!(structure_options & FREESASA_INCLUDE_HYDROGEN) && std::string(atom.type_symbol) == "H") {
+                continue;
+            }
+
+            // Pick the first alternative conformation for an atom
+            if (currentAltId != '.' && currentAltId != 'A') {
+                continue;
+            }
+
+            freesasa_structure_add_cif_atom(structure, &atom, classifier, structure_options);
+        }
+    }
+    return structure;
+}
+
 
-            if (models.count(model) == 0) {
+
+static freesasa_structure*
+structure_from_doc(const gemmi::cif::Document &doc,
+                         const std::string &filter,
+                         const int &col, 
+                         const freesasa_classifier *classifier,
+                         int structure_options)
+{
+    std::cout << "Creating a filtered freesasa structure !" << std::endl;
```

Comment:
Only strictly necessary things should be in the output I think. So I think the `cout`s should be deleted or inside `#ifdef DEBUG`

Reply:
Sorry I meant to delete all these! I missed one. good catch. 
Ill start using `#DEFINE DEBUG`.